packages/mobile/src/send/SendAmount.tsx

```diff
@@ -180,14 +201,14 @@ export class SendAmount extends React.Component<Props, State> {
   }
 
   getConfirmationInput = (type: TransactionTypes) => {
-    const amount = parseInputAmount(this.state.amount)
+    const dollarsAmount = this.getDollarsAmount()
     const recipient = this.getRecipient()
     // TODO (Rossy) Remove address field from some recipient types.
     const recipientAddress = getAddressFromRecipient(recipient, this.props.e164NumberToAddress)
 
     const confirmationInput: ConfirmationInput = {
       recipient,
-      amount,
+      amount: new BigNumber(dollarsAmount.toPrecision(18, BigNumber.ROUND_UP)),
```

Comment:
what's going on here?

Reply:
Ah yes 😸  It definitely needs explaining.

This is needed to have the maximum supported precision when the initial amount was entered in the local currency. Otherwise the confirmation screen won't display the same local currency amount that was entered or the fee/send will fail because of a decimal part remaining.

Let's take an example.
- user enters `2` MXN (their local currency)
- user expects `2` MXN to be displayed in the confirmation screen
- technically the confirmation screens receives `2 / exchangeRate`, if we take `19.67` as the exchange rate, the dollar amount is then `0.10167768174885612608`. Given we want to display the local amount in the confirmation too, we need to convert again, `0.10167768174885612608 * 19.67 = 1.9999999999999999999936` which displays as `1.99 MXN` (which is not what the user expects) and also yields the following fee calculation error: `CalculateFee/Error calculating fee:invalid number value (arg="value", coderType="uint256", value="101677681748856126.08")` as `0.10167768174885612608` is converted to wei but still has a decimal part which is not accepted by the contract.
- that's why we limit the precision to 18 decimals and round up at this precision! This really only affects local currency. If the user enters `2.99` cUSD, it still displays `2.99` in the confirmation screen.

I thought about passing the original local amount to the confirmation screen but this complicates things and we would still have the problem once we display the transaction in the activity feed which doesn't have the original entered amount.

Let me know if you see an alternative. Also I think we should be able to derive the `18` magic number from a constant. Maybe from `WEI_PER_CELO`? Or is there another place where its already defined?

Reply:
I'd prefer we just pass around all the information we need in it's original BigNumber form instead of mixing in this complicated decimal handling.

You're closer certainly to this feature than me so I'lll leave it to you to decide. If you do want to leave this rounding logic in place, please make a utility for it.

Reply:
I think this is the best place for now to handle this. As this is the screen where the local currency amount is entered and is the one dealing with the conversion.

This input is also used for the request confirmation screen, and passes the right value to handle the case I described.

I'll wrap it in a function and add the full comment above.